### PR TITLE
Fix build and proof failures for CBMC TCP proofs

### PR DESCRIPTION
**Pull request by tony-josi-aws**
This PR fixes the build issues and proof failures for the TCP CBMC proofs post the IPv6 changes.

Test Steps
-----------
```
cd test/cbmc/proofs/
python run-cbmc-proofs.py
```
Related Issue
-----------
Failing CBMC proofs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_TCP_Reception.c [hunk 1]
```diff
@@ -74,7 +74,7 @@
  * Skip past TCP header options when doing Selective ACK, until there are no
  * more options left.
  */
-        static void prvReadSackOption( const uint8_t * const pucPtr,
+        _static void prvReadSackOption( const uint8_t * const pucPtr,
```
[the remark this answers is not in the document]
Comment (tony-josi-aws) [reply source/FreeRTOS_TCP_Reception.c]: updated

File: test/cbmc/proofs/TCP/prvTCPPrepareSend/TCPPrepareSend_harness.c [hunk 2]
```diff
@@ -62,12 +62,26 @@ NetworkBufferDescriptor_t * pxGetNetworkBufferWithDescriptor( size_t xRequestedS
...
 {
-    FreeRTOS_Socket_t * pxSocket = ensure_FreeRTOS_Socket_t_is_allocated();
+    FreeRTOS_Socket_t * pxSocket = malloc( sizeof( FreeRTOS_Socket_t ) );
+    __CPROVER_assume( pxSocket != NULL );
+    pxSocket->u.xTCP.rxStream = malloc( sizeof( StreamBuffer_t ) );
+    __CPROVER_assume( pxSocket->u.xTCP.rxStream != NULL );
+    pxSocket->u.xTCP.txStream = malloc( sizeof( StreamBuffer_t ) );
+    __CPROVER_assume( pxSocket->u.xTCP.txStream != NULL );
+    pxSocket->u.xTCP.pxPeerSocket = malloc( sizeof( FreeRTOS_Socket_t ) );
+    __CPROVER_assume( pxSocket->u.xTCP.pxPeerSocket != NULL );
```
[the remark this answers is not in the document]
Comment (tony-josi-aws) [reply test/cbmc/proofs/TCP/prvTCPPrepareSend/TCPPrepareSend_harness.c]: fixed

File: test/cbmc/proofs/TCP/prvTCPPrepareSend/TCPPrepareSend_harness.c [hunk 3]
```diff
@@ -62,12 +62,24 @@ NetworkBufferDescriptor_t * pxGetNetworkBufferWithDescriptor( size_t xRequestedS
...
+    FreeRTOS_Socket_t * pxSocket = malloc( sizeof( FreeRTOS_Socket_t ) );
+    __CPROVER_assume( pxSocket != NULL );
+    pxSocket->u.xTCP.rxStream = malloc( sizeof( StreamBuffer_t ) );
+    pxSocket->u.xTCP.txStream = malloc( sizeof( StreamBuffer_t ) );
+    pxSocket->u.xTCP.pxPeerSocket = malloc( sizeof( FreeRTOS_Socket_t ) );
+
     NetworkBufferDescriptor_t * pxNetworkBuffer = ensure_FreeRTOS_NetworkBuffer_is_allocated();
     size_t socketSize = sizeof( FreeRTOS_Socket_t );
-    size_t bufferSize = sizeof( TCPPacket_t );
+    /* Allocates min. buffer size required for the proof */
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/TCP/prvTCPPrepareSend/TCPPrepareSend_harness.c:81]: This comment just tells me what is written in the code right below.

Ideally, comments should answer questions like why does the proof need exactly `sizeof( TCPPacket_t ) + uxIPHeaderSizeSocket( pxSocket )` and not just `sizeof( TCPPacket_t )`.

We could improve on these comments.

**Review by AniruddhaKanhere — APPROVED** [review]
Approved with reservation about comments.

**Pull request merged**